components/compliance-service/compliance.go

```diff
@@ -333,37 +334,133 @@ func getManagerConnection(connectionFactory *secureconn.Factory,
+		esSidecarClient   es_sidecar.EsSidecarClient
+		recurrence        *rrule.RRule
+		purgeJobName      = "purge"
+		purgeScheduleName = "periodic_purge"
+	)
+
 	if conf.ComplianceReportDays > 0 {
-		purgePolicies = append(purgePolicies, dlsserver.PurgePolicy{
-			IndexName:          fmt.Sprintf("comp-%s-s", mappings.ComplianceCurrentTimeSeriesIndicesVersion),
-			PurgeOlderThanDays: conf.ComplianceReportDays,
-		})
-		purgePolicies = append(purgePolicies, dlsserver.PurgePolicy{
-			IndexName:          fmt.Sprintf("comp-%s-r", mappings.ComplianceCurrentTimeSeriesIndicesVersion),
-			PurgeOlderThanDays: conf.ComplianceReportDays,
-		})
+		desiredPurgePolicies = defaultPurgePolicies
 	}
 
-	var err error
-	var esSidecarConn *grpc.ClientConn
 	if os.Getenv("RUN_MODE") == "test" {
 		logrus.Infof(`Skipping ES sidecar-service dial due to RUN_MODE env var being set to "test"`)
 	} else {
 		timeoutCtx, cancel := context.WithTimeout(ctx, 60*time.Second)
 		defer cancel()
-		// Data Lifecycle Interface
+
 		logrus.Infof("Connecting to %s", conf.ESSidecarAddress)
 		esSidecarConn, err = connFactory.DialContext(timeoutCtx, "es-sidecar-service",
 			conf.ESSidecarAddress, grpc.WithBlock())
 		if err != nil || esSidecarConn == nil {
 			logrus.WithFields(logrus.Fields{"error": err}).Fatal("Failed to create ES Sidecar connection")
 			return nil, err
 		}
+		esSidecarClient = es_sidecar.NewEsSidecarClient(esSidecarConn)
+	}
+
+	err = purge.ConfigureManager(
+		purgeJobName, purgeScheduleName, cerealManager, esSidecarClient,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to configure %s workflow", purgeJobName)
 	}
-	return dlsserver.NewDataLifecycleManageableServer(es_sidecar.NewEsSidecarClient(esSidecarConn), purgePolicies), nil
+
+	recurrence, err = rrule.NewRRule(rrule.ROption{
+		Freq:     rrule.DAILY,
+		Interval: 1,
+		Dtstart:  time.Now(),
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not create recurrence rule for %s", purgeScheduleName)
+	}
+
+	enabled := false
+	if len(desiredPurgePolicies.Es) > 0 {
+		enabled = true
+	}
+	err = cerealManager.CreateWorkflowSchedule(
+		purgeScheduleName,
+		purgeJobName,
+		desiredPurgePolicies,
+		enabled,
+		recurrence,
+	)
+
+	if err == cereal.ErrWorkflowScheduleExists {
+		// If the workflow already exists it may have different indices configured
+		// for our policies. If they differ then we'll need to update the workflow.
+		sched, err := cerealManager.GetWorkflowScheduleByName(
+			ctx, purgeScheduleName, purgeJobName,
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get purge schedule from job manager")
+		}
+
+		currentPolicies := purge.Policies{
+			Es: map[string]purge.EsPolicy{},
+		}
+		if err = sched.GetParameters(&currentPolicies); err != nil {
+			return nil, errors.Wrap(err, "failed to get purge policies from job manager")
+		}
+
+		for name, dp := range defaultPurgePolicies.Es {
+			p, ok := currentPolicies.Es[name]
+			if ok {
```

Comment:
Yes. The current implementation only includes policies that are desired, so if a user doesn't want the policy configured it's removed from the workflow. It's impossible to tell if they've removed it on purpose or not. We could add an `enabled` field to each policy and manage it and always include every policy in the workflow parameters, but that's not how it works right now.

Reply:
Thinking about this some more I'm going to write some words to make clear my thinking on it:

* The "default policies" represent both the policies that can be updated or enabled, and the default values of those policies.
* The "current policies" represent policies that are enabled.
* We do a one time migration from configuration to "current policies", after which "current policies" are managed through the servers `Configure()` endpoint.
* We always migrate the "current policies" index at startup in case there has been a migration.

This means that any new additional default policies will never be added to the "current policies" unless it's added through the `Configure()` endpoint.

Even if we always included all policies in the "current policies" we'd have no idea if was enabled by the user or was enabled by default.

So, if we want to support "adding new current policies that are enabled" and we want to do it safely then each policy will need to have a "disabled" field that we'll set explicitly during migration or configuration and all policies will be set in current policies. Then the actions would have to respect that field.

Reply:
Okay, it's been rewritten to properly migrate previous config, handle new policies, and update existing policies if they have an index or custom field change.